File: Gemini 1.5/Python/2-4.py
def binary_search(arr, target):
    """
    Performs binary search on a sorted list to find the index of a target value.

    Args:
      arr: A sorted list of integers.
      target: The value to search for.

    Returns:
      The index of the target value if found, otherwise -1.
    """
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return mid  # Target found
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1  # Target not found


if __name__ == '__main__':
    arr = [2, 5, 7, 8, 11, 15]
    target = 11

    index = binary_search(arr, target)

    if index != -1:
        print(f"Target {target} found at index {index}")
    else:
        print(f"Target {target} not found!")

File: Gemini 1.5/C++/2-4.c
#include <iostream>
#include <vector>

using namespace std;

int binarySearch(vector<int>& arr, int target) {
    int left = 0;
    int right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            return mid; // Target found
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1; // Target not found
}

int main() {
    vector<int> arr = {2, 5, 7, 8, 11, 15};
    int target = 11;

    int index = binarySearch(arr, target);

    if (index != -1) {
        cout << "Target " << target << " found at index " << index << endl;
    } else {
        cout << "Target " << target << " not found!" << endl;
    }

    return 0;
}

